FRENCH/src/ch07-00-managing-growing-projects-with-packages-crates-and-modules.md

```diff
@@ -0,0 +1,120 @@
+-->
+
+# Gérer des projets grandissants avec les paquets, crates et modules
+
+<!--
+As you write large programs, organizing your code will be important because
+keeping track of your entire program in your head will become impossible. By
+grouping related functionality and separating code with distinct features,
+you’ll clarify where to find code that implements a particular feature and
+where to go to change how a feature works.
+-->
+
+Lorsque vous commencez à écrire des gros programmes, organiser votre code va
+devenir important car vous ne pourrez plus vous souvenir de l'intégralité de
+votre programme de tête. En regroupant des fonctionnalités qui ont des points
+communs et en séparant des autres fonctionnalités, vous clarifiez l'endroit où
+trouver le code qui implémente une fonctionnalité spécifique et où aller pour
+changer une fonctionnalité.
+
+<!--
+The programs we’ve written so far have been in one module in one file. As a
+project grows, you can organize code by splitting it into multiple modules and
+then multiple files. A package can contain multiple binary crates and
+optionally one library crate. As a package grows, you can extract parts into
+separate crates that become external dependencies. This chapter covers all
+these techniques. For very large projects of a set of interrelated packages
+that evolve together, Cargo provides workspaces, which we’ll cover in the
+[“Cargo Workspaces”][workspaces]<!-- ignore -- > section in Chapter 14.
+-->
+
+Le programme que nous avons écrit précédemment était dans un module et dans un
+seul fichier. Quand le projet grandit, vous pouvez organiser votre code en le
+découpant en plusieurs modules et ensuite en plusieurs fichiers. Un paquet peut
+contenir plusieurs crates binaires et accessoirement une crate de bibliothèque.
+Lorsque un paquet grandit, vous pouvez en extraire des parties dans des crates
+séparées qui deviennent des dépendances externes. Ce chapitre va aborder toutes
+ces techniques. Pour un projet de très grande envergure qui a des paquets
+interconnectés qui évoluent ensemble, Cargo propose les espaces de travail, que
+nous allons découvrir dans une section du
+[chapitre 14][workspaces]<!-- ignore -->.
+
+<!--
+In addition to grouping functionality, encapsulating implementation details
+lets you reuse code at a higher level: once you’ve implemented an operation,
+other code can call that code via the code’s public interface without knowing
+how the implementation works. The way you write code defines which parts are
+public for other code to use and which parts are private implementation details
+that you reserve the right to change. This is another way to limit the amount
+of detail you have to keep in your head.
+-->
+
+En plus de pouvoir regrouper des fonctionnalités, l'encapsulation des éléments
+implémentés vous permet de réutiliser votre code à un niveau supérieur : lorsque
+vous avez implémenté une opération, un autre code peut appeler ce code via
```

Comment:
This is unclear, the sentence should be reorganized and the translation should move away from English

Reply:
I don't find a way to say it better, if you have a better wording, i'm on !

Reply:
What about:

```markdown
En plus de regrouper des fonctionnalités, les modules vous permettent d'encapsuler les
détails de l'implémentation d'une opération : vous pouvez écrire du code puis l'utiliser comme une
abstraction à travers l'interface de programmation publique du code [...]
```

Reply:
Yeah, this rewording is much better, thank you ! :wink: